table/tables/partition.go

```diff
@@ -1284,7 +1284,12 @@ func partitionedTableUpdateRecord(gctx context.Context, ctx sessionctx.Context,
 		// So this special order is chosen: add record first, errors such as
 		// 'Key Already Exists' will generally happen during step1, errors are
 		// unlikely to happen in step2.
-		err = t.GetPartition(from).RemoveRecord(ctx, h, currData)
+		deleteData := currData
+		if !t.Meta().IsCommonHandle && !t.Meta().PKIsHandle && len(deleteData) > len(t.Cols()) {
```

Comment:
> if !t.Meta().IsCommonHandle && !t.Meta().PKIsHandle && len(deleteData) > len(t.Cols()) {

IMHO it's tricky to remove the handle column in this way...Is there any existing code like this?

Reply:
I agree that it is tricky and I would not propose it if it would not been done like this [here](https://github.com/pingcap/tidb/blob/d63a5fd33dee1c6eef907ca7fae16972cc403557/executor/delete.go#L58-L66). And now when looking at the code in [deleteSingleTableByChunk](https://github.com/pingcap/tidb/blob/d63a5fd33dee1c6eef907ca7fae16972cc403557/executor/delete.go#L80-L131) I think I agree with @tiancaiamao and will try to change the protocol/caller instead.